### DISCUSSION TITLE
How do I enable Scroll on Mosh?

### DISCUSSION THREAD

**Discussion by iflowerml** [blinksh/blink#1933] [Q&A] [answered]
Does scroll work on mosh? How do I add scroll to Mosh? Does Mosh have the scroll feature?

**Comment by iflowerml** [comment, marked as answer]
Mosh does not support scroll, due to it only dumping changes between screens, like video rendering protocols. This is the reason why Mosh is so fast. How can I add scroll to Mosh? Since Mosh cannot scroll we recommend pairing it with a terminal multiplexer like `Tmux` or `Screen`. These allow buffering for scrolling and you have access to your previous command outputs. 
How to scroll on `Tmux`: 

- `Ctrl`+`b` then `[` then you can use Arrows. Press q to quit scroll mode.
- You can also press `Ctrl-b`+ `PgUp` to go straight to copy mode and scroll one page up.
OR
- To scroll with mouse wheel/ touchpad: In your `~/.tmux.conf` file add: `set -g mouse on`. Save and exit, Then:  `tmux source your-tmux-file`
 
How to scroll in `Screen`:

1. Press `ctrl`+`a` then press `Esc`. (check what your `ESC` is or assign it to a custom press in Settings>Keyboard) [Read more](https://github.com/blinksh/blink/discussions/1923)
2. Press the `Up` and `Down` arrows or `PgUp` and `PgDn` to scroll .
3. Press `ESC` again to exit scroll mode.

As an extra tip, you can auto launch your `Tmux` or `Screen` session automatically on Mosh.
Go to Setting>Hosts, choose your host, scroll down to Mosh section. In the command field enter: `tmux new -A -s ${my_session_name}` 

Mosh’s limitations; [read more](https://github.com/mobile-shell/mosh/issues/122).
Mosh with scrollback setup [refrence](https://medium.com/@toja/tip-using-mosh-with-scrollback-257a54a848b3).

**Comment by tiagoefreitas** [comment]
@iflowerml 

I tried this but it doesn't work well with the finger scroll because blink translates touch to arrows in mosh, instead of mouse wheel. blink uses scroll in bink shell and in ssh, if you do the same for mosh it would work with tmux mouse mode.

I guess you decided on arrows in mosh because mosh does not have an history buffer and this is a good decision if tmux mouse mode did not exist 
but I don't really need the arrows mapped to touch because I have the arrows icon that does that same...its a waste 😦
